test/integration/reachability.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Reachability tests
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/sync/errgroup"
+)
+
+var (
+	// accessLogs is a mapping from app name to a list of request ids that should be present in it
+	accessLogs map[string][]string
+	accessMu   sync.Mutex
+)
+
+func testBasicReachability() error {
+	log.Printf("Verifying basic reachability across pods/services (a, b, and t)..")
+
+	accessLogs = make(map[string][]string)
+	for app := range pods {
```

Comment:
Just a thought for another CL, but it would probably be good to move away from the use of global variables in these tests, particularly now that the test is split across multiple files. We can define a testInstance struct to carry fields around and make most of these test functions methods on that struct.